addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/BAE091xSensorThingHandler.java

```diff
@@ -0,0 +1,249 @@
+ */
+@NonNullByDefault
+public class BAE091xSensorThingHandler extends OwBaseThingHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BAE091X);
+
+    private final Logger logger = LoggerFactory.getLogger(BAE091xSensorThingHandler.class);
+    private OwSensorType sensorType = OwSensorType.UNKNOWN;
+
+    public static final Set<OwSensorType> SUPPORTED_SENSOR_TYPES = Collections.singleton(OwSensorType.BAE0910);
+
+    public BAE091xSensorThingHandler(Thing thing, OwDynamicStateDescriptionProvider dynamicStateDescriptionProvider) {
+        super(thing, dynamicStateDescriptionProvider, SUPPORTED_SENSOR_TYPES);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            if (channelUID.getId().startsWith(CHANNEL_DIGITAL)) {
+                Bridge bridge = getBridge();
+                if (bridge != null) {
+                    OwserverBridgeHandler bridgeHandler = (OwserverBridgeHandler) bridge.getHandler();
+                    if (bridgeHandler != null) {
+                        if (!((BAE0910) sensors.get(0)).writeChannel(bridgeHandler, channelUID.getId(), command)) {
+                            logger.debug("writing to channel {} in thing {} not permitted (input channel)", channelUID,
+                                    this.thing.getUID());
+                        }
+                    } else {
+                        logger.warn("bridge handler not found");
+                    }
+                } else {
+                    logger.warn("bridge not found");
+                }
+            }
+        }
+        // TODO: PWM channels
+        super.handleCommand(channelUID, command);
+    }
+
+    @Override
+    public void initialize() {
+        if (!super.configureThingHandler()) {
+            return;
+        }
+
+        sensors.add(new BAE0910(sensorId, this));
+
+        scheduler.execute(() -> {
+            configureThingChannels();
+        });
+    }
+
+    @Override
+    protected void configureThingChannels() {
+        ThingUID thingUID = getThing().getUID();
+        logger.debug("configuring sensors for {}", thingUID);
+
+        BAE091xHandlerConfiguration configuration = getConfig().as(BAE091xHandlerConfiguration.class);
+
+        Set<OwChannelConfig> wantedChannel = new HashSet<>();
+
+        // Pin1:
+        switch (configuration.pin1) {
+            case CONFIG_BAE_PIN_DISABLED:
+                break;
+            case CONFIG_BAE_PIN_COUNTER:
+                wantedChannel.add(new OwChannelConfig(CHANNEL_COUNTER, CHANNEL_TYPE_UID_BAE_COUNTER));
+                break;
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "unknown configuration option for pin 1");
+                return;
+        }
+
+        // Pin2:
+        switch (configuration.pin2) {
```

Comment:
Is it possible to express this switch case in a simpler way? A lot of cases look very similar and I have the impression that a static configuration lookup table for those pins makes more sense?